### DISCUSSION TITLE
RC200 emulation

### DISCUSSION THREAD

**Discussion by ghost20219721** [emsesp/EMS-ESP32#1551] [Ideas]
Hi all,
MichaelDvP, as we discussed in discord, I am sending you the log of my system with RC200 for emulation
[log.txt](https://github.com/emsesp/EMS-ESP32/files/13889006/log.txt)

If it helps, the thermostat mode is manual, the selected temperature is 23.5C, and the room temperature is 23.6 - 23.7C. heating circuit 2

RC200 in my system has the following values
Type: Thermostat
Name: RC200/CW100
Brand Device ID: 0x39
Product ID: 157
Version: 41.08
 
If you need more information you can find me on discord.

**Comment by MichaelDvP** [comment]
Ok, first try with dummy RC200 on device-id 0x3A for hc3. Check with https://github.com/MichaelDvP/EMS-ESP32/releases/tag/test
If it does not work: Do you can configure the real RC200 to hc3 and log the working configuration?

**Reply by ghost20219721** [reply]
Thanks for the answer.
unfortunately it didn't work
[log-2.txt](https://github.com/emsesp/EMS-ESP32/files/13891683/log-2.txt)
New RC200 Log as HC3 Controller

but 1 note
when the real RC200 is present in the EMS-ESP control panel, it contains one entity named "Room temperature hcX from remote device", while the dummy RC200 has no entity at all. perhaps this is a problem due to which the main controller does not see the temperature (on the LCD - - - instead of the temperature value) and as a result produces errors: 1. Incorrect remote control (A11/3113), 2. Defected room temperature sensor (A11/3093) and 3. Humidity detected at feed pipe (A33/3133).

**Reply by MichaelDvP** [reply]
Found a bug, the remote sended with hc3 address, but hc1 telegram number. Try with new build, maybe this was the only cause.
The RC200 telegram is:
`thermostat(0x3A) -W-> thermostat(0x10), RemoteTemp(0x042D), data: 00 ED 09 47 01`
for the remote we use only the first 2 bytes (temperature 00ED=23.7°C). The 0947 seems to be a high resolution value, repeated in master telegram offset 22:
`thermostat(0x10) -W-> Me(0x0B), RC300Monitor(0x02A7), data: 00 ED 23 28 00 00 28 28 05 A0 02 03 01 00 4F 03 AD 00 00 11 01 02 09 47 00`
Maybe we need to add it, but the last byte (01) is unknown.

**Reply by ghost20219721** [reply]
Cool, now I can see entity and current temperature from remote sensor, and even can control HC3 thermostat from HA,  but RC310 continuously shows A33/3133 and may be all others as described in my previous post.. For now I'm at work and can't see all errors, only last one in esp dashboard

**Reply by ghost20219721** [reply]
My bad
the instructions have a more complete description of error A33/3113
Moisture detected in the supply pipe (humidity sensor activated, jumper MD1 of the mixer module is closed)

this circuit was used as a constant temperature and of course MD1 is shorted in my system :-)

in a couple of hours I’ll check without a jumper on the MD1 mixer contact and give feedback

**Reply by MichaelDvP** [reply]
To check if there are other errors you can use telnet and `read 10 12` That gives a set of error messages, 12 bytes each error, starting with 2 ASCII codes, next 2 bytes with error code, next date infos. Error 33/3133 will show 33 33 0C 3D, etc. But i think the 11/xxxx was due hc3 emulation sending hc1 telegram.

**Reply by ghost20219721** [reply]
OK, now hc3 works as need, but A11/3113 error still appears every 2 minutes
from manual "Check the configuration
Contradictory remote control configurations are recorded. For example, set RC100, actual RC200"
may be somwhere in the code there is a part from RC100H ?

**Reply by MichaelDvP** [reply]
What have you configuered as control? What control options do you have? Maybe we can go back to RC100H afer the bug in telegram-type is fixed?

**Reply by ghost20219721** [reply]
RC100H not available in the configuration of my RC310. There is only RC100, RC200, RC310.
![image](https://github.com/emsesp/EMS-ESP32/assets/156210993/ca41f034-e743-431a-afac-e6feb5b4f422)
Hc3 configured with RC200 control device and weather compensated control mode

**Reply by ghost20219721** [reply]
![IMG_4403](https://github.com/emsesp/EMS-ESP32/assets/156210993/a4cd93d7-2667-4152-a9d4-025d8912fae2)
only this options available on my RC310

**Reply by MichaelDvP** [reply]
I think we have two options, switch version back to RC100H and you select RC100 (without sending remothum) and try.
If this does not work we can send the temperature telegram with 5 bytes, but i don't know what the last byte means.
I've pushed the test build with first option, please try.

**Reply by ghost20219721** [reply]
Error A11/3113 still there.
RC100 as controller selected

**Reply by MichaelDvP** [reply]
I triy now to read the control settings and use RC200/RC100H version and telegram style dependend on setting. You have to choose the control first before sending the remotetemp. RC200 sends 5 bytes, last is always 1, don't know if this changes something (mode/dhw/?).

**Reply by ghost20219721** [reply]
so now I need to cold restart all heating system, select RC100 as controller for HC3 and only then send remote temp value, is this correct ? 

about last byte I think sets mode AUTO/Manual, don't now RC100 control only manual mode or it can work as AUTO ( day/night )

**Reply by MichaelDvP** [reply]
Why cold start heating system? Just restart ems-esp to remove old remotes from mem after changing the control. But a obsolete RC100H registered to emsesp does not have any effect to heating system, it only keeps showing in emsesp.
If your hc-control is set to RC200 the emsesp should emulate a RC200 with 5 bytes telegram, 
for RC100/RC100H it identifies as RC100H (i dont know RC100 version infos) and sends only temperature in the telegram.

**Reply by ghost20219721** [reply]
Restart ems-esp didnt help.. error still there.
It appears and disappears continiusly every 1-2 minutes
Remotetemp value publishing every 1m, 15secs, 30secs..

**Reply by MichaelDvP** [reply]
Can you take a log with the RC200 remote emulation, we have the log with real RC200 in https://github.com/emsesp/EMS-ESP32/discussions/1551#discussioncomment-8084891, there should be a difference.

**Reply by ghost20219721** [reply]
[log (1).txt](https://github.com/emsesp/EMS-ESP32/files/13927435/log.1.txt)
Emulated RC200 log

**Reply by MichaelDvP** [reply]
This was not wth the latest changes, it's still 2 bytes temperature message. Please update to my test (version is the same as before, but new commits). https://github.com/MichaelDvP/EMS-ESP32/releases/tag/test

**Reply by ghost20219721** [reply]
[log (2).txt](https://github.com/emsesp/EMS-ESP32/files/13927600/log.2.txt)
Updated

**Reply by MichaelDvP** [reply]
I think hc2 (0x39) is a real RC200, hc3 (0x3A) the emulation?
this looks ok: 
```
thermostat(0x3A) -W-> thermostat(0x10), RemoteTemp(0x042D), data: 00 D2 08 34 01
thermostat(0x39) -W-> thermostat(0x10), RemoteTemp(0x042C), data: 00 EB 09 33 01
```
There is no interview of the master to the remote, i don't know what generates the error. I think it's still the A11/3113?
I see a bug in anserweing ems+ telegrams, fixed the length check and remove queries to batterie/calibration from RC200 registration.
Check with new version.

**Reply by ghost20219721** [reply]
You are right hc2 (0x39) is a real RC200, hc3 (0x3A) the emulation
Error still on the screen, but something changed.
After last firmware upgrade this error appear and never disappear ( before it was ~2 min cycle ) 

New log 
[log.txt](https://github.com/emsesp/EMS-ESP32/files/13928058/log-3.txt)
[log2.txt](https://github.com/emsesp/EMS-ESP32/files/13928062/log.3.txt)
two logs in different time 
in log2 you can see error A11/3113 appears after MD1 jumper opened and error A33 cleared

**Reply by MichaelDvP** [reply]
I can see the error appear, but no communication with the 0x3A remote that could cause it. The real RC200 have a second id iin the version info, maybe we need this, i've added to latest version now. 
The master thermostat shows the roomtemp as reported by the remote. Seems to be workig, only the error should not show up.

**Reply by ghost20219721** [reply]
You are right! error not showing now.. But after firmware upload and restart it was.. in my case physically restart thermostat RC310 ( remove from wall bracket  and attach it back ) helps, no errors any more. 38 minutes normal flight.

**Reply by MichaelDvP** [reply]
Have you seen any effect of the 5.th byte fixed to 1 in the temperature telegram? 
Should we try to send without it?

**Reply by ghost20219721** [reply]
First of all now I can say that everything is working well, no errors for 18 hours! That's perfect.
About 5th byte I didn't see any changes, it was working before you add it, but with error A11.
We can try to disable it and I will test on my system.
I think this byte is something like temp correction or working mode, but I can mistake.
For example I can switch real RC200 to auto mode and send you logs.

**Reply by ghost20219721** [reply]
[log-RC200_auto_mode.txt](https://github.com/emsesp/EMS-ESP32/files/13931003/log-4.txt)

**Reply by ghost20219721** [reply]
`thermostat(0x39) -W-> thermostat(0x10), RemoteTemp(0x042C), data: 00 E7 09 0A 01`
it still there, nothing changed.

**Comment by JakeFromTheDark** [comment]
I had a similar issue with Rego 3000, solved by changing EMS-ESP / Settings / Application / EMS Tx Mode from EMS to EMS+ followed by Heat Pump Off+sleep+On.
